Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -593,6 +593,8 @@ def update_download_hops(self, download, new_hops):
         """
         infohash = binascii.hexlify(download.tdef.get_infohash())
         self._logger.info("Updating the amount of hops of download %s", infohash)
+        pstate = download.get_persistent_download_config()
+        pstate.set('state', 'engineresumedata', (yield download.save_resume_data()))
```

Comment:
What if the torrent is in the state where it's still fetching metadata? Does the `save_resume_data` fire with `None` then?

Reply:
Yep, it returns `None`. In this case libtorrent will give us a `save_resume_data_failed_alert`, which will trigger the errback being called, but since the `Failure` is then trapped in `_on_resume_err`, we end up with `None`.